### PR TITLE
API and web socket updates.

### PR DESCRIPTION
**Pull request by kbenne**
This PR introduces new APIs for:

`/tests` to show running tests
`/testcases/<namespace>` to put test cases into namespaces
`/users/<username>/testcases` for user specific test cases 

Also a new WebSocket API. See tests https://github.com/NREL/boptest-service/blob/issue/24-websocket/test/test_websocket_api.py

### CODE REVIEW

**Comment by kbenne** [comment]
@tijcolem the environment has changed so we'll need to talk about config changes. I'd like to deploy this to dev for testing.

**Comment by kbenne** [comment]
Also, @tijcolem, we might need some nginx changes to proxy the websocket. https://www.nginx.com/blog/websocket-nginx/

**Comment by kbenne** [comment]
I need to update the GitHub action to get it passing. The provision script changed a little.

**Comment by kbenne** [comment]
This PR will require a coordinated deployment with the dashboard v1.2 tag https://github.com/NREL/boptest-dashboard/releases/tag/v1.2

File: test/test_websocket_api.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+from requests_toolbelt import MultipartEncoder
+from collections import OrderedDict
+import asyncio
+import websockets
+import pytest
+import json
+import uuid
+from pytest_check import check
+
+host = os.environ.get("BOPTEST_SERVER", "http://web")
```
Comment (tijcolem) [inline test/test_websocket_api.py:11]: We may want to change BOPTEST_SERVER to just the domain name? I noticed the websockets.connect("ws://web") is currently hard coded and uses different protocol "ws" vs "http" 

BOPTEST_SERVER=api.dev.boptest.net

Comment (kbenne) [reply test/test_websocket_api.py:11]: how about BOPTEST_SERVER and BOPTEST_WS_SERVER so that we can config http versus https?

Comment (tijcolem) [reply test/test_websocket_api.py:11]: right, forgot about HTTPS.  I like those two envs

**Pull request merged**